### DISCUSSION TITLE
Encoder class (org.owasp.esapi.reference.DefaultEncoder) CTOR threw exception

### DISCUSSION THREAD

**Discussion by wtpau** [ESAPI/esapi-java-legacy#696] [Q&A] [answered]
Hi All,

I just upgraded my project from esapi 2.2.0.0 to 2.4.0.0 and I encounter an error "Encoder class (org.owasp.esapi.reference.DefaultEncoder) CTOR threw exception"

Here is the code that throw the exception when it run to the "return ESAPI.encoder().encodeForSQL(ORACLE_CODEC, param);" line.

```
package my.sample;

import org.owasp.esapi.ESAPI;
import org.owasp.esapi.codesc.Codec;
import org.owasp.esapi.codecs.OracleCodec;

public class SanitizeUtility {
 private static final Codec<?> ORACLE_CODEC = new OracleCodec();

public static String sanitizeSQLParam(String param) {
return ESAPI.encoder().encodeForSQL(ORACLE_CODEC, param);
}
}
```

Just wondering anyone know what the issue could be? How could you fix it?

Many Thanks!
Wai Pau

**Comment by kwwall** [comment, marked as answer]
Did you remember to update your ESAPI.Logger class in your ESAPI.properties file and other related changes as discussed in https://github.com/ESAPI/esapi-java-legacy/blob/develop/documentation/esapi4java-core-2.2.1.0-release-notes.txt#L128-L155 and several successive release notes? Check that first.

**Reply by wtpau** [reply]
Thanks Kevin,

I got it working now by adding 2 lines to my ESAPI.properties

```
Logger.UserInfo=true
Logger.ClientInfo=true
```

Wai
